pymaster/field.py

```diff
@@ -658,3 +681,176 @@ def get_ell_sampling(self):
+            it will default to ``4*lmax``.
+        spin (:obj:`int`): Spin of this field. If ``None`` it will
+            default to 0 or 2 if ``field`` contains 1 or 2 arrays,
+            respectively.
+        beam (`array`): Spherical harmonic transform of the instrumental beam
+            (assumed to be rotationally symmetric - i.e. no :math:`m`
+            dependence) associated with this field. If ``None``, no beam will
+            be corrected for. Otherwise, this array should have at least of
+            size ``lmax+1``.
+        field_is_weighted (:obj:`bool`): Set to ``True`` if the input field has
+            already been multiplied by the source weights.
+        lonlat (:obj:`bool`): If ``True``, longitude and latitude in degrees
+            are provided as input. If ``False``, colatitude and longitude in
+            radians are provided instead.
+    """
+    def __init__(self, positions, weights, field, lmax, lmax_mask=None,
+                 spin=None, positions_rand=None, weights_rand=None, beam=None,
+                 field_is_weighted=False, lonlat=False):
+        # 0. Preliminary initializations
+        if ut.HAVE_DUCC:
+            self.sht_calculator = 'ducc'
+        else:
+            raise ValueError("DUCC is needed, but currently not installed.")
+
+        # These first attributes are compulsory for all fields
+        self.lite = True
+        self.mask = None
+        self.beam = None
+        self.n_iter = None
+        self.n_iter_mask = None
+        self.pure_e = False
+        self.pure_b = False
+        self._Nw = 0.
+        self._Nf = 0.
+        self._alpha = None
+        self.ainfo = ut.NmtAlmInfo(lmax)
+
+        # The remaining attributes are only required for non-lite maps
+        self.maps = None
+        self.temp = None
+        self.alm_temp = None
+        self.minfo = None
+        self.n_temp = 0
+        do_clustering = positions_rand is not None and weights_rand is not None
+
+        # Can only have field or do clustering
+        if field is not None and do_clustering:
+            raise ValueError("If positions_rand and weights_rand are "
+                             "provided, field must be None.")
+
+        # Sanity checks on positions and weights
+        positions = np.array(positions, dtype=np.float64)
+        weights = np.array(weights, dtype=np.float64)
+        pw_list = [(positions, weights)]
+        if do_clustering:
+            positions_rand = np.array(positions_rand, dtype=np.float64)
+            weights_rand = np.array(weights_rand, dtype=np.float64)
+            pw_list += [(positions_rand, weights_rand)]
+        for p, w in pw_list:
+            if np.shape(p) != (2, len(w)):
+                raise ValueError("Positions must be 2D array of shape"
+                                 " (2, len(weights)).")
+            if lonlat:
+                p[0] = np.radians(90. - p[1])
+                p[1] = np.radians(p[0])
```

Comment:
Ah, but you just changed `p[0]`!!!

Reply:
Right, stupid bug. I only wonder why the unit tests didn't pick this up before (since this bug has been there before). We should double check those too.

Reply:
OK, I have the answer: we shouldn't be using `np.radians(ang_deg)` because it takes as input the whole real axis. We should rather use `np.pi/180.*ang_deg` so we can meaningfully unit test for wrong lonlat input.